kernel/fs/iso9660.inc

```diff
@@ -736,3 +736,1044 @@ cd_compare_name:
+        mov     ecx, edi
+        call    iso9660_GetFileInfo.copy_file_info
+
+        ; copy encoding
+        movzx   eax, byte[esp + 20]
+        mov     [edi + 4], eax
+
+        add     edi, 40
+;-----------------------------------------------------------------------------
+        ; copy name
+        push    ebx
+        push    edi
+        push    eax
+
+        add     ebx, ISO9660_DIRECTORY_RECORD.name
+        mov     esi, ebx
+        mov     cl, [ebx - 1]
+        cmp     cl, 1
+        jne     @f
+        cmp     byte[ebx], 0
+        jne     @f
+        mov     dword[edi], '.'
+        jmp     .end_copy
+@@:
+        cmp     cl, byte[ebx]
+        jne     @f
+        mov     dword[edi], '..'
+        cmp     eax, 2
+        jne     .end_copy
+        mov     dword[edi], ('.' shl 16) + '.'
+        mov     word[edi+ 2], 0
+        jmp     .end_copy
+@@:
+;-----------------------------------------------------------------------------
+        ; copy real name
+        movzx   ecx, byte[ebx - 1]
+        cmp     [ebp + ISO9660.type_encoding], 0
+        jnz     .ucs2
+        cmp     eax, 2
+        je      .ascii2utf16
+@@:
+        sub     ecx, 1 ; CF + ZF
+        jbe     @f
+        movsd
+        cmp     byte[esi], ';'
+        jne     @b
+@@:
+        mov     byte[edi], 0
+        jmp     .end_copy
+;-----------------------------------------------------------------------------
+.ascii2utf16:
+        lodsb
+        call    ansi2uni_char
+        stosw
+        cmp     byte[esi], ';'
+        je      @f
+        loop    .ascii2utf16
+@@:
+        mov     word[edi], 0
+        jmp     .end_copy
+;-----------------------------------------------------------------------------
+.ucs2:
+        shr     ecx, 1
+        cmp     eax, 1
+        ja      .ucs2utf16
+        ; convert ucs2 to ascii
+@@:
+        lodsw
+        cmp     ax, 0x3B00
+        je      @f
+        xchg    al, ah
+        call    uni2ansi_char
+        stosb
+        loop    @b
+@@:
+        mov     byte[edi], 0
+        jmp     .end_copy
+;-----------------------------------------------------------------------------
+.ucs2utf16:
+        cmp     edx, 2 ; utf16 encoding
+        jne     .ucs2utf8
+        ; convert ucs2 to utf16LE
+@@:
+        lodsw
+        cmp     ax, 0x3B00
+        je      @f
+        xchg    al, ah
+        stosw
+        loop    @b
+@@:
+        mov     word[edi], 0
+        jmp     .end_copy
+;-----------------------------------------------------------------------------
+.ucs2utf8:
+        ; convert ucs2 to utf8
+        mov     ebx, ecx
+        shl     ecx, 1
+@@:
+        lodsw
+        cmp     ax, 0x3B00
+        je      @f
+        xchg    ah, al
+        call    UTF16to8
+        dec     ebx
+        jnz     @b
+@@:
+        mov     byte[edi], 0
+;-----------------------------------------------------------------------------
+.end_copy:
+        pop     eax
+        pop     edi
+        pop     ebx
+        add     edi, 264
```

Comment:
magic number

Reply:
fixed

Reply:
these values are not defined to the kernel code

Reply:
> these values are not defined to the kernel code

but you can comment what do they mean